clients/src/main/java/org/apache/kafka/common/utils/ByteUtils.java

```diff
@@ -0,0 +1,327 @@
+            }
+        }
+        value |= b << i;
+        return (value >>> 1) ^ -(value & 1);
+    }
+
+    /**
+     * Write the given integer following the variable-length zig-zag encoding from
+     * <a href="http://code.google.com/apis/protocolbuffers/docs/encoding.html">
+     * Google Protocol Buffers</a> into the output.
+     *
+     * @param value The value to write
+     * @param out The output to write to
+     */
+    public static void writeVarint(int value, DataOutput out) throws IOException {
+        value = (value << 1) ^ (value >> 31);
+        while ((value & 0xffffff80) != 0L) {
+            out.writeByte((value & 0x7f) | 0x80);
+            value >>>= 7;
+        }
+        out.writeByte((byte) value);
+    }
+
+    /**
+     * Write the given integer following the variable-length zig-zag encoding from
+     * <a href="http://code.google.com/apis/protocolbuffers/docs/encoding.html">
+     * Google Protocol Buffers</a> into the output.
+     *
+     * @param value The value to write
+     * @param buffer The output to write to
+     */
+    public static void writeVarint(int value, ByteBuffer buffer) {
+        value = (value << 1) ^ (value >> 31);
+        while ((value & 0xffffff80) != 0L) {
+            byte b = (byte) ((value & 0x7f) | 0x80);
+            buffer.put(b);
+            value >>>= 7;
+        }
+        buffer.put((byte) value);
+    }
+
+    /**
+     * Write the given integer following the variable-length zig-zag encoding from
+     * <a href="http://code.google.com/apis/protocolbuffers/docs/encoding.html">
+     * Google Protocol Buffers</a> into the output.
+     *
+     * @param value The value to write
+     * @param out The output to write to
+     */
+    public static void writeVarlong(long value, DataOutput out) throws IOException {
+        value = (value << 1) ^ (value >> 63);
+        while ((value & 0xffffffffffffff80L) != 0L) {
+            out.writeByte(((int) value & 0x7f) | 0x80);
+            value >>>= 7;
+        }
+        out.writeByte((byte) value);
+    }
+
+    /**
+     * Write the given integer following the variable-length zig-zag encoding from
+     * <a href="http://code.google.com/apis/protocolbuffers/docs/encoding.html">
+     * Google Protocol Buffers</a> into the output.
+     *
+     * @param value The value to write
+     * @param buffer The buffer to write to
+     */
+    public static void writeVarlong(long value, ByteBuffer buffer) {
+        value = (value << 1) ^ (value >> 63);
+        while ((value & 0xffffffffffffff80L) != 0L) {
+            byte b = (byte) (((int) value & 0x7f) | 0x80);
+            buffer.put(b);
+            value >>>= 7;
+        }
+        buffer.put((byte) value);
+    }
+
+    /**
+     * Number of bytes needed to encode an integer in variable-length format.
+     *
+     * @param value The signed value
+     */
+    public static int sizeOfVarint(int value) {
```

Comment:
My bad, the parameter is just an `int` and not a buffer. Thanks to @hachikuji for correcting me.